redbot/core/bot.py

```diff
@@ -582,6 +582,7 @@ def colour(self) -> NoReturn:
 
         async with self._config.custom(SHARED_API_TOKENS, service_name).all() as group:
             group.update(tokens)
+        self.dispatch("red_api_tokens_update", service_name, group)
```

Comment:
This should be changed to either dispatch a mappingproxy or just the service name (indicating to cogs they may need to re-fetch the data if they use that service)

I'd like to avoid issues with dispatching a mutable object here.